dataactvalidator/filestreaming/csvAbstractReader.py

```diff
@@ -67,25 +66,54 @@ def openFile(self,region,bucket,filename,csvSchema,bucketName,errorFilename):
             raise ResponseException("Error in header row: CSV file must use only '|' or ',' as the delimiter", StatusCode.CLIENT_ERROR, ValueError, ValidationError.headerError)
 
         self.delimiter = "|" if line.count("|") != 0 else ","
-        for row in csv.reader([line],dialect='excel', delimiter=self.delimiter):
-            for cell in row :
-                headerValue = FieldCleaner.cleanString(cell)
-                if( not headerValue in possibleFields) :
+
+        # check to see if header contains long or short column names
+        # TODO: is there a better way to do this?
+        validation_db = ValidatorValidationInterface()
+        longNameDict = validation_db.getLongToShortColname()
+        headerNames = line.split(self.delimiter)
+        colMatches = 0
+        for i, value in enumerate(headerNames):
+            if value in longNameDict:
+                colMatches += 1
+        if colMatches > 15:
```

Comment:
Actually I just realized a problem with the majority thing, if more than half of the fields present are these extra fields we're allowing but ignoring, this will choose short names.  That said, if a user submits a file with short names, but some of those extra fields are the long name, checking for presence of a set number could also make the wrong choice.  Possibly we could count both presence of short and long names and go with whichever is greater (or pick one in case of a tie).  I may also be overthinking this, we could just state that those extra fields we're not using are not allowed to be the other version of a field name (in which case we could flag that as a header error).